src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -76,11 +80,27 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     }
 
     String user = userService.getCurrentUser().getEmail();
-    String text = Jsoup.clean(request.getParameter("text"), Whitelist.none());
+    String userText =
+        Jsoup.clean(request.getParameter("text"), Whitelist.none());
 
-    Message message = new Message(user, text);
-    datastore.storeMessage(message);
+    String regex = "(https?://\\S+\\.(png|jpg))";
```

Comment:
You should put this in the class scope and declare it as `private static final REGEX = "(...)";`.